extensions/interactions/LogicProof/static/js/student.js

```diff
@@ -994,12 +994,16 @@ var logicProofStudent = (function() {
   *          }
   */
   var buildProof = function(proofString, questionInstance) {
-    if (proofString.slice(-1) === '\n') {
-      proofString = proofString.slice(0, proofString.length - 1);
-    }
     var lineStrings = proofString.split('\n');
+    // Ignore blank lines at the end

```

Comment:
I agree with the principal. However extra blank lines at the end are a common issue and don't really qualify as a mistake by the student. Extra blank lines in the middle are not really desirable, so if submitted there is a warning message to the student and they can fix them.